### PR TITLE
Add logic for `2048`

### PR DESCRIPTION
**Pull request by Rishiraj0100**

### CODE REVIEW

File: interactions/buttons/games.py [hunk 1]
```diff
@@ -1178,6 +1182,23 @@ def number_to_emoji(self):
             GameString += "".join(row) + "\n"
         return GameString
 
+    def lost(self):
+        if self.has_empty:
+            return
+
+        board = [[b for b in i] for i in self.board]
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1189]: **Unnecessary looping.**
```py
>>> board
[[0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3]]
>>> [[b for b in i] for i in board]
[[0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3]]
>>>
```
Directly assign `board = self.board`

File: interactions/buttons/games.py [hunk 2]
```diff
@@ -1236,7 +1257,9 @@ async def upward(self, button: discord.ui.Button, interaction: discord.Interacti
                 url="https://cdn.discordapp.com/attachments/894938379697913916/922771882904793120/41NgOgTVblL.png"
             )
         )
-
+        if game.lost():
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1260]: `game` not defined. Must be `self.game`

File: interactions/buttons/games.py [hunk 3]
```diff
@@ -1274,6 +1297,9 @@ async def left(self, button: discord.ui.Button, interaction: discord.Interaction
                 url="https://cdn.discordapp.com/attachments/894938379697913916/922771882904793120/41NgOgTVblL.png"
             )
         )
+        if game.lost():
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1300]: `game` not defined. Must be `self.game`

File: interactions/buttons/games.py [hunk 4]
```diff
@@ -1300,6 +1326,9 @@ async def down(self, button: discord.ui.Button, interaction: discord.Interaction
...
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1329]: `game` not defined. Must be `self.game`

File: interactions/buttons/games.py [hunk 5]
```diff
@@ -1326,6 +1355,9 @@ async def right(self, button: discord.ui.Button, interaction: discord.Interactio
...
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1358]: `game` not defined. Must be `self.game`

File: interactions/buttons/games.py [hunk 6]
```diff
@@ -1166,9 +1166,13 @@ def spawn_new(self):
             (j, i) for j, sub in enumerate(board) for i, el in enumerate(sub) if el == 0
         ]
         if not zeroes:
+            self.has_empty = False
             return
         i, j = random.choice(zeroes)
         board[i][j] = 2
+        self.has_empty = True
+
+        if all([0 not in i for i in self.board]):self.has_empty = False
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1175]: **Unnecessary comprehension**
```py
import itertools
self.has_empty = 0 in itertools.chain(*self.board)
```

**Review by rtk-rnjn — CHANGES_REQUESTED** [review]
Fix the issues I mentioned

File: interactions/buttons/games.py (hunk 1 above)
Comment (Rishiraj0100) [reply interactions/buttons/games.py:1189]: Cuz changing the self.board updates board var

File: interactions/buttons/games.py (hunk 3 above)
Comment (Rishiraj0100) [reply interactions/buttons/games.py:1300]: Hn whi h

**Comment by Rishiraj0100** [comment]
> Fix the issues I mentioned

Sry bro I can't, u merge and do or make changes and merge I did `allow edits from maintainer` so u can

**Comment by rtk-rnjn** [comment]
OK! I will see later

File: interactions/buttons/games.py [hunk 7]
```diff
@@ -1188,6 +1191,23 @@ def number_to_emoji(self):
...
+
+        self.MoveUp()
+        if self.board != board: return restore()
+        self.MoveDown()
+        if self.board != board: return restore()
+        self.MoveLeft()
+        if self.board != board: return restore()
+        self.MoveRight()
+        if self.board != board: return restore()
+        return True
```
Comment (rtk-rnjn) [inline interactions/buttons/games.py:1209]: These lines makes me scare. L

**Review by rtk-rnjn — APPROVED** [review]

File: interactions/buttons/games.py (hunk 7 above)
Comment (Rishiraj0100) [reply interactions/buttons/games.py:1209]: Wdym?

**Comment by rtk-rnjn** [comment]
ok! I am merging for now! not implementing both DeepSource and CodeFactor are screaming. I hope this won't break the bot

**Pull request merged**